java/parquet/src/main/java/sleeper/io/parquet/record/RecordWriteSupport.java

```diff
@@ -50,6 +47,8 @@ public void prepareForWrite(RecordConsumer recordConsumer) {
     }
 
     public void write(Record record) {
-        recordWriter.write(record);
+        if (null != recordWriter) {
+            recordWriter.write(record);
+        }
```

Comment:
With a null check, it won't throw any exception if it's not initialised properly. Maybe Objects.requireNonNull would be better?